### PR TITLE
Updated to Fastify v4, dropped error handler

### PR DESCRIPTION
**Pull request by mcollina**
Signed-off-by: Matteo Collina <hello@matteocollina.com>

This PR updates to Fastify v4 and drops the custom error handler support.
I do not think that's needed with Fastify v4 by default as that's now way simpler to manage.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,7 +8,8 @@
 Defaults for Fastify that everyone can agree on™.<br>
 This plugin adds some useful utilities to your Fastify instance, see the API section to learn more.
 
-Supports Fastify versions `3.x`.
+Supports Fastify versions `4.x`.
+Please refer to [this branch](https://github.com/fastify/fastify-sensible/tree/4.x) and related versions for Fastify `3.x` compatibility.
```
Comment (Fdawgs) [inline README.md:12]: Should this be https://github.com/fastify/fastify-sensible/tree/3.x?

Comment (mcollina) [reply README.md:12]: no because current release is on the v4.x line for Fastify v3, we renamed the module.

Comment (Fdawgs) [reply README.md:12]: Gotcha

**Comment by hsynlms** [comment]
May I ask for the details of dropping the custom error handler?
I wonder what Fastify v4 gives us different than v3 to achieve the same motivation.

**The motivation was:**
> This plugins also adds a custom error handler, which hides the error message in case of 500 errors that are being generated by throwed errors, instead it returns Something went wrong.

**Pull request merged**